openfe/setup/methods/rbfe_methods.py

```diff
@@ -0,0 +1,257 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+"""Contains RBFE methods
+
+This module implements.....
+
+"""
+from __future__ import annotations
+
+from openfe.setup import LigandAtomMapping, LigandMolecule
+from openfe.setup.methods import FEMethod
+from typing import Dict, Union
+from pydantic import BaseModel, validator
+from openff.units import unit
+
+# define a timestep
+# this isn't convertible to time (e.g ps) and is used to not confuse these two
+# definitions of "duration" within a simulation
+unit.define('timestep = [timestep] = _ = timesteps')
+
+
+class ForcefieldSettings(BaseModel):
+    """Settings describing the forcefield to use for each component
+
+    Attributes
+    ----------
+    forcefield
+      a mapping of each components name to the xml forcefield to apply
+    nonbonded_electrostatics
+      which nonbonded electrostatics method to use, currently only 'PME'
+      allowed
+    nonbonded_cutoff
+      range of nonbonded forces
+    constraints
+      which bonds to apply constraints to
+    rigid_water
+      whether apply rigid constraints to water molecules, default True
+    """
+    class Config:
+        arbitrary_types_allowed = True
+
+    # mapping of component name to forcefield path
+    forcefield: Dict[str, str]
+    nonbonded_electostatics = 'PME'
+    nonbonded_cutoff = 0.9 * unit.nanometer
+    constraints = 'HBonds'
+    rigid_water = True
+
+
+class StatePoint(BaseModel):
+    """Description of the temperature and pressure to model"""
+    class Config:
+        arbitrary_types_allowed = True
+
+    temperature = 298.15 * unit.kelvin
+    pressure = 1 * unit.bar
+
+    @validator('pressure', 'temperature')
+    def must_be_positive(cls, v):
+        if v <= 0:
+            raise ValueError("Pressure and temperature must be positive")
+        return v
+
+    @validator('pressure')
+    def is_pressure(cls, v):
+        if not v.is_compatible_with(unit.bar):
+            raise ValueError("Must be pressure value, e.g. use unit.bar")
+        return v
+
+    @validator('temperature')
+    def is_temperature(cls, v):
+        if not v.is_compatible_with(unit.kelvin):
+            raise ValueError("Must be temperature value, e.g. use unit.kelvin")
+
+
+class LambdaProtocolSettings(BaseModel):
+    """Settings for the lambda protocol
+
+    This describes a fixed number of windows
+
+    Attributes
+    ----------
+    functions : str, default 'default'
+      key of which switching functions to use for alchemical mutation
+    windows : int, default 11
+      number of lambda windows to calculate
+    sample_endstate : bool, default False
+      whether to sample the endstates ???
+    """
+    functions = 'default'
+    windows = 11
+    sample_endstates = False
+
+
+class MonteCarloBarostatSettings(BaseModel):
```

Comment:
Something to think about - what if we want to use MonteCarloAnisotropicBarostat instead?